### ISSUE TITLE
Form does not trigger onSubmit in Reat Native Expo

### ISSUE THREAD

**Issue by slandrei** [tamagui/tamagui#1716]
### Current Behavior

Form does not trigger onSubmit in Reat Native Expo

### Expected Behavior

onSubmit function should be triggered on click submit button inside Form

### Tamagui Version

```markdown
1.72.1
```

### Platform (Web, iOS, Android)

```markdown
Android (and maybe Ios, not tested).
```

### Reproduction

```markdown
import { View as RNView, StyleSheet } from "react-native";
import React, { useState } from "react";
import { View, Text, Form, Button, Input, H3, Label, Checkbox } from "tamagui";
import { Link } from "expo-router";

const Login = () => {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [rememberMe, setRememberMe] = useState(false);

    const submitForm = () => {
        console.log("Triggered");
        console.log(email, password, rememberMe);
    };

    return (
        <RNView style={styles.formContainer}>
            <Form gap={15} onSubmit={submitForm}>
                <H3 textAlign='center' marginBottom={100}>
                    Welcome back
                </H3>
                <Input
                    size={"$4"}
                    placeholder='Email'
                    value={email}
                    onChangeText={setEmail}
                />
                <Input
                    size={"$4"}
                    placeholder='Password'
                    value={password}
                    onChangeText={setPassword}
                />

                <View
                    fd='row'
                    alignItems='center'
                    justifyContent='space-between'
                    px={15}
                >
                    <View fd='row' alignItems='center'>
                        <Checkbox
                            size='$4'
                            mr={15}
                            checked={rememberMe}
                            onCheckedChange={setRememberMe}
                        >
                            <Checkbox.Indicator>
                                {/* Icon for checked */}
                                <Text>x</Text>
                            </Checkbox.Indicator>
                        </Checkbox>
                        <Label>Remember me</Label>
                    </View>
                </View>

                <Form.Trigger mt={30} asChild>
                    <Button>Login</Button>
                </Form.Trigger>
            </Form>
        </RNView>
    );
};

export default Login;

const styles = StyleSheet.create({
    formContainer: {
        flex: 1,
        justifyContent: "center",
        paddingHorizontal: 15,
        paddingVertical: 20,
    },
});
```

### System Info

```markdown
System:
    OS: Windows 10 10.0.19045
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Memory: 2.77 GB / 7.96 GB
  Binaries:
    Node: 18.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (117.0.2045.43)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @babel/core: ^7.20.0 => 7.23.0
    @expo/webpack-config: ^19.0.0 => 19.0.0
    @react-navigation/drawer: ^6.6.4 => 6.6.4
    @tamagui/config: ^1.72.1 => 1.72.1
    @tamagui/font-inter: ^1.72.1 => 1.72.1
    @tamagui/font-silkscreen: ^1.72.1 => 1.72.1
    @tamagui/shorthands: ^1.72.1 => 1.72.1
    @tamagui/themes: ^1.72.1 => 1.72.1
    @types/react: ~18.2.14 => 18.2.22
    expo: ~49.0.12 => 49.0.12
    expo-constants: ~14.4.2 => 14.4.2
    expo-device: ~5.4.0 => 5.4.0
    expo-font: ~11.4.0 => 11.4.0
    expo-linking: ~5.0.2 => 5.0.2
    expo-router: ^2.0.0 => 2.0.8
    expo-splash-screen: ~0.20.5 => 0.20.5
    expo-status-bar: ~1.6.0 => 1.6.0
    expo-updates: ~0.18.13 => 0.18.13
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    react-native: 0.72.5 => 0.72.5
    react-native-gesture-handler: ~2.12.0 => 2.12.1
    react-native-reanimated: ~3.3.0 => 3.3.0
    react-native-safe-area-context: 4.6.3 => 4.6.3
    react-native-screens: ~3.22.0 => 3.22.1
    react-native-web: ~0.19.6 => 0.19.9
    tamagui: ^1.72.1 => 1.72.1
    typescript: ^5.1.3 => 5.2.2
```

**Comment by Jean-MarcCheynier** [comment]
up! experiencing the same

**Comment by lewxdev** [comment]
Just bumped our packages from `1.31.2` to `1.72.3` and experiencing the same issue on iOS

**Comment by ckeyes88** [comment]
I came across this as well on the `^1.74.1`. The button calls the `onPress` fine but won't trigger the form submit.

**Comment by asyndesis** [comment]
+1 Same issue IOS

<img width="214" alt="Screenshot 2023-09-30 at 9 15 06 PM" src="https://github.com/tamagui/tamagui/assets/1721012/6151debc-2aed-4bb8-a862-26a766ea393a">

**Comment by Jean-MarcCheynier** [comment]
I found out there is really little test on tamagui. This bug would be caught immediately with components testing

**Comment by natew** [comment]
fix in 0605e21def72ac1fad4fca4fa57abc778282ad9a going out now in v1.74.3

@Jean-MarcCheynier yes our web tests are very robust, our native tests are less so. we have maestro set up with some tests, and we unit test a variety of things (something like 300 unit tests run every release covering many areas native uses).

it's an ongoing process. we have improved a lot since beginning of the year, and we're setting up maestro cloud on an OSS sponsorship so we can run more native tests in CI.
